src/SpinnerKernel.java

```diff
@@ -0,0 +1,168 @@
+import components.map.Map;
+import components.map.Map2;
+
+public class SpinnerKernel<K, V> extends Map2<K, V> {
+    //NOTES: Within the sphere of Map, is there a way
```

Comment:
Don't extend map. Just use it as your representation, such as with the following line:
```java
private Map<String, Integer> map;
```

Then set the value in the constructor like:
```java
this.map = new Map2<String, Integer>();
```